app/controllers/CompanyInformationController.scala

```diff
@@ -45,6 +47,20 @@ class CompanyInformationController @Inject() (
     }
   }
 
+  def getCompanyInformationV2: Action[AnyContent] = authorisedRequest async {
+    implicit request: RequestWithEori[AnyContent] =>
+      val eori = request.eori.value
+
+      companyInformationRepository.get(eori).flatMap {
+        case Some(companyInformation) => Future.successful(Ok(Json.toJson(companyInformation)))
+        case None                     =>
+          retrieveCompanyInformation(eori).map {
```

Comment:
Better to move this in a separate private method (name could be, retrieveCompanyInfo and existing retrieveCompanyInfo could be renamed to companyInfoFromSub09AndStoreInRepo (or some other meaningful name) to make it more readable and avoid the nested case statements.

And new private method could also be used (if you are happy to make change) in getCompanyInformation.

Reply:
I have kept the same name but moved all the retrieval logic into the private method.